server/src/main/java/com/senlainc/warsaw/tyurin/dao/OrderDAO.java

```diff
@@ -27,17 +39,465 @@ public void setInstance() {
     }
 
     @Override
-    public void addOrder(Order order) {
-        orders.add(order);
+    public void addOrder(Order order) throws Exception {
+        long orderStatusId = getOrderStatusId(order.getOrderStatus().toString());
+        try(Connection connection = dbConnector.createConnection();
+            PreparedStatement ordersInsert = connection.prepareStatement("insert into orders (price, " +
+                    "submission_date, start_date, completion_date, garage_place_id, order_status_id) " +
+                    "values (?, ?, ?, ?, ?, ?);");
+            PreparedStatement orderCraftsmanInsert = connection.prepareStatement("insert into order_craftsman (order_id, craftsman_id) " +
+                    "values (?, ?);")){
+            connection.setAutoCommit(false);
+            ordersInsert.setDouble(1, order.getPrice());
+            ordersInsert.setTimestamp(2, Timestamp.valueOf(order.getSubmissionDate()));
+            ordersInsert.setTimestamp(3, Timestamp.valueOf(order.getStartDate()));
+            ordersInsert.setTimestamp(4, Timestamp.valueOf(order.getCompletionDate()));
+            ordersInsert.setLong(5, order.getGaragePlaceId());
+            ordersInsert.setLong(6, orderStatusId);
+            ordersInsert.executeUpdate();
+
+            order.getCraftsmenId().forEach(craftsmanId -> {
+                try {
+                    orderCraftsmanInsert.setLong(1, order.getId());
+                    orderCraftsmanInsert.setLong(2, craftsmanId);
+                    orderCraftsmanInsert.executeUpdate();
+                } catch (SQLException exception) {
+                    exception.printStackTrace();
+                }
+            });
+            connection.commit();
+            connection.setAutoCommit(true);
+        } catch (Exception exception) {
+            throw new Exception("Can't add order");
+        }
+    }
```

Comment:
Thanks. The only one method use transactional support

Reply:
Is it only one place you need transaction management?

Reply:
Yes, in other situations I need to update only one table per time

Reply:
Ok. Good then, go ahead!

Reply:
Thank you!